commons/src/main/java/com/powsybl/commons/reporter/LoggerTreeReporterDeserializer.java

```diff
@@ -0,0 +1,148 @@
+/**
+ * Copyright (c) 2021, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.commons.reporter;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonToken;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.*;
+
+/**
+ * @author Florian Dupuy <florian.dupuy at rte-france.com>
+ */
+public class LoggerTreeReporterDeserializer extends StdDeserializer<LoggerTreeReporter> {
+
+    LoggerTreeReporterDeserializer() {
+        super(LoggerTreeReporter.class);
+    }
+
+    @Override
+    public LoggerTreeReporter deserialize(JsonParser parser, DeserializationContext ctx) throws IOException {
+        String taskKey = null;
+        String defaultName = "";
+        Map<String, Object> taskValues = new HashMap<>();
+        List<Report> reports = new ArrayList<>();
+        List<LoggerTreeReporter> childReporters = new ArrayList<>();
+
+        while (parser.nextToken() != JsonToken.END_OBJECT) {
+            switch (parser.getCurrentName()) {
+                case "version":
+                    parser.nextToken(); // skip
+                    break;
+
+                case "taskKey":
+                    taskKey = parser.nextTextValue();
+                    break;
+
+                case "defaultName":
+                    defaultName = parser.nextTextValue();
+                    break;
+
+                case "taskValues":
+                    parser.nextToken();
+                    taskValues = parser.readValueAs(new TypeReference<HashMap<String, Object>>() {
+                    });
+                    break;
+
+                case "reports":
+                    parser.nextToken();
+                    reports = parser.readValueAs(new TypeReference<ArrayList<Report>>() {
+                    });
+                    break;
+
+                case "childReporters":
+                    parser.nextToken();
+                    childReporters = parser.readValueAs(new TypeReference<ArrayList<LoggerTreeReporter>>() {
```

Comment:
here you create children of type LoggerTreeReporter but they could have been any ReportSeeker before the export